ariadne/cardano/src/Ariadne/Wallet/Cardano/Kernel/DB/AcidState.hs

```diff
@@ -117,6 +118,11 @@ instance Buildable NewPendingError where
     build (NewPendingFailed npf) =
         bprint ("NewPendingFailed " % build) npf
 
+instance Show NewPendingError where
+    show = formatToString build
+
+instance Exception NewPendingError
+
 newPending :: HdAccountId
```

Comment:
What does this do?

Reply:
If this comment is about `HdAccountId` argument, I have the same question. If not, I have a question about this argument :)
Does it mean that currently `newPending` assumes only one input account? If yes, then it's bad, because it's possible to send a transaction from multiple accounts in our case.

Reply:
> Does it mean that currently newPending assumes only one input account?

Yes. This issue is a bug, and it is a bit hard to work around. Currently this `HdAccountId` is chosen as the account that owns the change address. The purpose of this parameter is to specify which account checkpoints to use for storing the pending transaction. One of the things that is done when inserting a new pending transaction is checking that all of its inputs are available in the account that the transaction belongs to.

Hence merging this PR in its current state will break sending transactions from multiple accounts.

Reply:
> Hence merging this PR in its current state will break sending transactions from multiple accounts.

Okay, then I think we should merge in its current state and there are two options:
1. Implement it properly in this PR.
2. Comment out _something_, so that `newPending` is still called, but does nothing. Create an issue to implement it properly (with proper description).